clippy_lints/src/methods/is_digit_ascii_radix.rs

```diff
@@ -0,0 +1,43 @@
+//! Lint for `c.is_digit(10)`
+
+use super::IS_DIGIT_ASCII_RADIX;
+use clippy_utils::{
+    consts::constant_full_int, consts::FullInt, diagnostics::span_lint_and_sugg, source::snippet_with_applicability,
+};
+use rustc_errors::Applicability;
+use rustc_hir::Expr;
+use rustc_lint::LateContext;
+
+pub(super) fn check<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &'tcx Expr<'_>,
+    self_arg: &'tcx Expr<'_>,
+    radix: &'tcx Expr<'_>,
+) {
+    if !cx.typeck_results().expr_ty_adjusted(self_arg).peel_refs().is_char() {
+        return;
+    }
+
+    if let Some(radix) = constant_full_int(cx, cx.typeck_results(), radix) {
+        let replacement = match radix {
+            FullInt::S(10) | FullInt::U(10) => "is_ascii_digit",
+            FullInt::S(16) | FullInt::U(16) => "is_ascii_hexdigit",
+            _ => return,
+        };
+        let mut applicability = Applicability::MachineApplicable;
+
+        span_lint_and_sugg(
+            cx,
+            IS_DIGIT_ASCII_RADIX,
+            expr.span,
+            "use of `char::is_digit(..)` with literal radix of 10 or 16",
```

Comment:
Since you mention the used radix, you could go one step further and only name the one been using. For that, you would store the found value and then use format, like so:

```rs
&format!("use of `char::is_digit(..)` with literal radix of {}", radiax_int),
```

or maybe:

```rs
&format!("use of `char::is_digit({})`", radiax_int),
```

This is optional, the current lint message already works well, and I'm also happy if you want to keep it this way :upside_down_face: